src/main/java/com/example/auctionapp/service/UserService.java

```diff
@@ -27,5 +37,43 @@ public UserDetails loadUserByEmail(String email){
         return loadUserByUsername(email);
     }
 
+    public ResponseEntity<?> updateUser(UpdateRequest updateRequest) {
+        User user = userRepository.getById(updateRequest.getId());
+        user.setName(updateRequest.getName());
+        user.setSurname(updateRequest.getSurname());
+        user.setEmail(updateRequest.getEmail());
+        user.setGender(updateRequest.getGender());
+        user.setDateOfBirth(updateRequest.getDateOfBirth());
+        user.setPhoneNumber(updateRequest.getPhoneNumber());
+        user.setPhoto(updateRequest.getPhoto());
+        if (updateRequest.getId() != null &&
```

Comment:
I believe there is no need for id check here. If it was null, user fetching would fail on ln 41, and code would not get to this point